api/src/paths/project/{projectId}/survey/{surveyId}/update.ts

```diff
@@ -273,7 +273,14 @@ export const getSurveyProprietorData = async (
 
   const response = await connection.query(sqlStatement.text, sqlStatement.values);
 
-  return (response && response.rows && response.rows[0] && new GetSurveyProprietorData(response.rows[0])) || null;
+  const result =
+    (response && response.rows && response.rows[0] && new GetSurveyProprietorData(response.rows[0])) || null;
+
+  if (!result) {
+    throw new HTTP400('Failed to get project survey proprietor data');
+  }
+
+  return result;
```

Comment:
I think you were on vacation, but this was changed to intentionally to return null if there is no proprietor data.  So we don't want to throw an error here.